[WIP] Switch CircleCI template to CircleCI 2.0.

felicianotech opened the pull request:
Fixes #52.

This is a first draft at converting the CircleCI template from using CircleCI 1.0 to 2.0

Notes:

1. I haven't looked into optimizing the config yet. Just trying to do a basic conversion.
2. The PHP Docker images are locked to PHP vX.Y, not a specific bug release. We can change that if the project owners decide they specifically want a full vX.Y.Z.
3. I don't know Mustache well so this config may not be proper syntax for Mustache.
4. I haven't run this config nor tested it yet. I plan to soon but others can join in as well. ;)
5. The new config file should be located at ./.circleci/config.yml instead of ./circle.yml. How would I get that to work here?

gitlost commented on README.md:
```diff
@@ -252,7 +252,7 @@ The following files are also included unless the `--skip-tests` is used:
 		options:
 		  - travis
-		  - circle
+		  - circleci
```
This shouldn't be changed I think for BC.

felicianotech replied:
BC?

gitlost replied:
Sorry, backward compatibility.

felicianotech replied:
Ah. Makes sense. I'll revert shortly.

felicianotech commented:
- rebased on latest master
- addressed comment about backwards compatibility
- added some more CircleCI 2.0 config changes to the actual project outside of the new config itself.

gitlost commented:
@travislopes thanks for trying it out, according to https://hub.docker.com/r/circleci/mysql/ the mysql user name is `root`.

felicianotech commented:
Added a comment addressing the latest comments here.

gitlost commented on templates/plugin-circle.mustache:
```diff
@@ -1,39 +1,112 @@
+          name: "Install Dependencies"
+          command: |
+            sudo apt-get update && sudo apt-get install subversion
+            sudo docker-php-ext-install mysqli
```
Thanks for the feedback @travislopes , stole that from  https://github.com/wp-cli/scaffold-command/pull/118. Not sure about the way it's installing `mysql-client-5.7` (which is missing from circleci docker image), seems kludgey...

felicianotech commented:
Awesome. Anything else needed from me?

gitlost commented:
Cool, if you're happy with it @felicianotech then no, will merge!

felicianotech commented:
Nope, LGTM though there is conflicts now.

felicianotech commented:
Okay I merged in master to fix the conflicts.

schlessera commented:
Thanks for the pull-request (and the patience), @felicianotech !
